### ISSUE TITLE
Weird behaviour with solve

### ISSUE THREAD

**Issue by weakit** [sympy/sympy#20211]
```python
>>> from sympy import *
>>> var('x')
x
>>> eq = Eq(x**5 + 2*x + 1, 0)

>>> solve(eq, x)  # this works
[CRootOf(x**5 + 2*x + 1, 0), CRootOf(x**5 + 2*x + 1, 1), CRootOf(x**5 + 2*x + 1, 2), CRootOf(x**5 + 2*x + 1, 3), CRootOf(x**5 + 2*x + 1, 4)]

>>> solve([eq], x)  # this doesn't for some reason?
[]
```

Shouldn't the result be the same, regardless of whether the Expression are sent in an iterable?

I was using `1.6.1`, but it's the same on the current master.

**Comment by oscarbenjamin** [comment]
Yes, the `solve` function is a mess. I started trying to reorganise it in #17993 but then I ended up sidetracked. Feel free to finish off some of the work from there.

**Comment by sayandip18** [comment]
@oscarbenjamin  is the goal is to ultimately deprecate `solve` and replace it with `solveset` or to keep both?

**Comment by oscarbenjamin** [comment]
I don't think that we should ever actually deprecate `solve` although we could hollow it out and rebuild it on top of other solvers. Right now none of the other solvers matches all of its capability.

**Comment by weakit** [comment]
So pretty much re-doing the entire thing?
